pandas/tests/extension/test_sparse.py

```diff
@@ -480,8 +480,7 @@ class TestParsing(BaseSparseTests, base.BaseParsingTests):
     def test_EA_types(self, engine, data):
         expected_msg = r".*must implement _from_sequence_of_strings.*"
         with pytest.raises(NotImplementedError, match=expected_msg):
-            with tm.assert_produces_warning(FutureWarning, match="astype from"):
-                super().test_EA_types(engine, data)
```

Comment:
How come this warning context is no longer needed?

(I see a lot of these type of changes in this PR but not 100% clear how they relate to your changes in `_warnings.py`)

Reply:
> Is it related to that☝️?

Yeah, the change in `_warnings.py` ensure `tm.assert_produces_warning` _still_ checks for a warning when the code in the context raises an exception.

For example if I add back `with tm.assert_produces_warning(FutureWarning, match="astype from"):` here I get

```
<traceback>

E               NotImplementedError: Extension Array: <class 'pandas.core.arrays.sparse.array.SparseArray'> must implement _from_sequence_of_strings in order to be used in parser methods

pandas/io/parsers/base_parser.py:809: NotImplementedError

During handling of the above exception, another exception occurred:

<more traceback>

E           AssertionError: Did not see expected warning of class 'FutureWarning'

pandas/_testing/_warnings.py:144: AssertionError
```

which is useful to see if `tm.assert_produces_warning` is still needed (and hence a lot of removals)